status-im/status-react
# [#15019] small option card component

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #15019 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)
Implements the [small option card component](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=16507%3A187682&t=5ZAyiMikBdLsKZVi-4) in its two variants: `main`  & `icon`.

### Review notes

A new category in the `Preview Quo2.0 Components` section has been added, it's called "onboarding":

Credit to @J-Son89  for d83a14a0c5952b3a88b57b4e45045b8e667bcd9e. He did a PR (not merged yet) and I took his `mock-fn`.

### Testing notes
Tests were added for this component. You can find them at `quo2.components.small-option-card.--tests--.small-option-card-component-spec`.

It's hard to check if the source's uri passed to a `fast-image` is correct. so I'm just testing that the `fast-image` is rendered.
These are the props that  `fast-image` with uri source `resources/images/mock/small_opt_card_main.png` has during testing:

```clojure
#js {:testID "small-option-card.main-image",
     :tintColor nil,
     :style #js {:position "absolute", 
                 :left 0,
                 :right 0,
                 :top 0,
                 :bottom 0},
     :source #js {:__packager_asset true,
                  :width 100,
                  :height 100,
                  :uri "file://assets/full/path/to/directory/icon@2x.png", 
                  :scale 2}, 
     :onFastImageLoadStart nil,
     :onFastImageProgress nil,
     :onFastImageLoad #object[Function],
     :onFastImageError #object[Function], 
     :onFastImageLoadEnd nil,
     :resizeMode "contain",
     :children nil}
```
That `uri` inside `source` key is not the same I provided, so I can't use those props to check it.


Please, share any ideas that you think would work to test if a `fast-image` component is rendered using a specific uri source.

#### Platforms

- Android
- iOS

#### Areas that maybe impacted


##### Functional


##### Non-functional

### Steps to test


- Open Status
- Navigate to Quo2 component previews -> cards -> small-option-card
- You should be able to  play with the components:

![image](https://user-images.githubusercontent.com/90291778/218587240-978982c8-7379-43f7-b3bf-67d0bc564976.png)
![image](https://user-images.githubusercontent.com/90291778/218587267-d913073b-debb-4ef1-bbb0-f3d61b1757f6.png)


status: ready
